### ISSUE TITLE
.oh-my-zsh/plugins/git/git.plugin.zsh:2: not an identifier: _omz_git_git_cmd^M

### ISSUE THREAD

**Issue by PendoNL** [robbyrussell/oh-my-zsh#5724]
Ever since the last update I get to see this when opening my terminal:

```
Last login: Thu Dec 22 21:45:12 on ttys000
/Users/pendo/.oh-my-zsh/plugins/git/git.plugin.zsh:2: not an identifier: _omz_git_git_cmd^M
[~]$
```

Not sure if more info is needed, I've got nothing else... thanks in advance.

**Comment by mcornella** [comment]
This is because you have core.autocrlf global set to something incompatible with zsh needing always LF only as end of line character. 

The temporary solution to that is changing your `core.autocrlf` git config setting to use `input` or `false`. See https://github.com/robbyrussell/oh-my-zsh/issues/4402#issuecomment-143976458.

The permanent fix would be merging #4071 but we need testers confirming that it works.

To sum up, you need to run the following:
```zsh
cd $ZSH
git config core.autocrlf false
git rm --cached -r .
git reset --hard
```

Don't worry, your error is the first time I've seen happen. The '^M' part of the message I've seen though, but now you know for next time. If you have extra time I'd like to ping you when something is ready to test to see if it bypasses the core.autoclrf incompatible settings.

Thanks!

**Comment by PendoNL** [comment]
If I can be of any help, let me know. Just have to be guided a bit I guess, since I'm not completely sure what autocrlf does or means. But thanks for your help!

I guess I should do the reset part for every repository that gives me errors when adding files/committing a message, right?

**Comment by mcornella** [comment]
It basically deals with how git resets the line endings of the fetched files. In this case shell files should all use LF ending, and with your current config some files get changed to CR EOLs which is the default in macos.

This has more info: https://help.github.com/articles/dealing-with-line-endings/

I have to change that PR significantly now that I've thought of a better alternative, but when I do I'll let you know how you can test it as well.

Thanks!
—
Marc
